lib/transaction/index.js

```diff
@@ -881,6 +881,13 @@ Transaction.prototype.acceptDistributedTraceHeaders = acceptDistributedTraceHead
 function acceptDistributedTraceHeaders(transportType, headers) {
   const transport = TRANSPORT_TYPES_SET[transportType] ? transportType : TRANSPORT_TYPES.UNKNOWN
 
+  if (typeof headers !== 'object') {
+    logger.trace(
+      'Ignoring distributed trace headers for transaction %s. Headers not passed in as object.',
+      this.id)
+    return
+  }
```

Comment:
Fun fact... in JavaScript typeof null is an 'object'.

```
> let blah = null
undefined
> typeof blah === 'object'
true
```

So you'll want to check null/undefined first. Would do something like `if (headers == null || typeof headers !== 'object')`.

Also, might as well put first prior to the transport code.